control-plane/csi-controller/src/main.rs

```diff
@@ -1,55 +1,82 @@
-use env_logger::{Builder, Env};
-use git_version::git_version;
-use structopt::StructOpt;
+use tracing::info;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Registry};
+
+use clap::{App, Arg};
 
 mod client;
-pub mod controller;
-pub mod identity;
-pub use client::{ApiClientError, MayastorApiClient};
+mod controller;
+mod identity;
+use client::{ApiClientError, MayastorApiClient};
 mod server;
 
-#[macro_use]
-extern crate log;
-extern crate env_logger;
-
 const CSI_SOCKET: &str = "/var/tmp/csi.sock";
 
-#[derive(Debug, Clone, StructOpt)]
-#[structopt(
-    name = "Mayastor CSI Controller",
-    about = "test",
-    version = git_version!(args = ["--tags", "--abbrev=12"], fallback="unkown"),
-    setting(structopt::clap::AppSettings::ColoredHelp)
-)]
-
-struct CsiControllerCliArgs {
-    #[structopt(short = "r", long = "rest-endpoint")]
-    pub rest_endpoint: String,
-    #[structopt(short="c", long="csi-socket", default_value=CSI_SOCKET)]
-    pub csi_socket: String,
-    #[structopt(short="l", long="log-level", default_value="info", possible_values=&["info", "debug", "trace"])]
-    pub log_level: String,
-}
+#[tokio::main(worker_threads = 2)]
+pub async fn main() -> Result<(), String> {
+    let args = App::new("Mayastor k8s pool operator")
+        .author(clap::crate_authors!())
+        .version(clap::crate_version!())
+        .settings(&[
+            clap::AppSettings::ColoredHelp,
+            clap::AppSettings::ColorAlways,
+        ])
+        .arg(
+            Arg::with_name("endpoint")
+                .long("rest-endpoint")
+                .short("-r")
+                .env("ENDPOINT")
+                .default_value("http://ksnode-1:30011")
+                .help("an URL endpoint to the mayastor control plane"),
+        )
+        .arg(
+            Arg::with_name("socket")
+                .long("csi-socket")
+                .short("-c")
+                .env("CSI_SOCKET")
+                .default_value(CSI_SOCKET)
+                .help("CSI socket path"),
+        )
+        .arg(
+            Arg::with_name("jaeger")
+                .short("-j")
+                .env("JAEGER_ENDPOINT")
+                .help("enable open telemetry and forward to jaeger"),
+        )
+        .get_matches();
```

Comment:
The new code looks a bit more verbose than existing.

Reply:
It is mike, but my experience has shown that over time, you start to struggle a bit with the attributes and .. well.. 

This change does not make the code any better not any worse so, I just prefer to be explicit than macro magic. Its a limitation of my brain :) Not in all cases ofcourse.